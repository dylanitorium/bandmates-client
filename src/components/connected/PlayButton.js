import { connect } from 'react-redux';
import * as interfaceActions from 'state/modules/interface/actions';
import PlayButton from '../pure/WaveForm/PlayButton/PlayButton';
import {
  audioSelector,
  isPlayingSelector,
} from 'state/selectors';

const mapStateToProps = state => ({
  audio: audioSelector(state),
  isPlaying: isPlayingSelector(state),
});

const mapDispatchToProps = {
  onPlayClick: interfaceActions.playToggle,
};

export default connect(mapStateToProps, mapDispatchToProps)(PlayButton);
