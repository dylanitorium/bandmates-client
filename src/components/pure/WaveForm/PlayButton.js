import React from 'react';
import Radium from 'radium';
import PropTypes from 'prop-types';
import Play from 'react-icons/lib/fa/play-circle-o';
import Pause from 'react-icons/lib/fa/pause-circle-o';

const buttonStyle = {
  border: 0,
  background: 'none',
  fontSize: '50px',
  color: 'white',
  opacity: 0.7,
  position: 'absolute',
  bottom: '15px',
  left: '10px',
  ':active': {
    outline: 'none',
  },
  ':focus': {
    outline: 'none',
  },
}

const PlayButton = props => (
  <button style={buttonStyle} onClick={props.onPlayClick}>
  {
    props.isPlaying ? (
      <Pause />
    ) : (
      <Play />
    )
  }
  </button>
);

PlayButton.propTypes = {
  isPlaying: PropTypes.bool.isRequired,
  onPlayClick: PropTypes.func.isRequired,
};

export default Radium(PlayButton);
