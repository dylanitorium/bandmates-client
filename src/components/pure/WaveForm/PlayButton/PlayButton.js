import React from 'react';
import PropTypes from 'prop-types';
import BasicAudioFile from 'utils/BasicAudioFile';
import styles from './play-button.css';

const PlayButton = props => (
  <div className={styles.button} onClick={props.onPlayClick}>
  {
    !props.audio ? (
      <i className={'fas fa-spinner fa-spin'}/>
    ) : !props.isPlaying ? (
      <i className={'far fa-play-circle'}/>
    ) : (
      <i className={'far fa-pause-circle'}/>
    )
  }
  </div>
);

PlayButton.propTypes = {
  audio: PropTypes.instanceOf(BasicAudioFile),
  isPlaying: PropTypes.bool.isRequired,
  onPlayClick: PropTypes.func.isRequired,
};

export default PlayButton;
