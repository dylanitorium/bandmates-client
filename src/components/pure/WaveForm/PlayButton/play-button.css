.button {
  border: 0;
  background: none;
  font-size: 50px;
  color: white;
  opacity: 0.7;
  position: absolute;
  bottom: 15px;
  left: 10px;
}

.button:active: {
  outline: none;
}

.button:focus: {
  outline: none;
}
