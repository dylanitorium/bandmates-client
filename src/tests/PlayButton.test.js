import React from 'react';
import renderer from 'react-test-renderer';
import PlayButton from '../components/pure/WaveForm/PlayButton/PlayButton';
import { playButton } from './fixtures';

describe('<PlayButton />', () => {
  test('renders a spinner when audio is loading', () => {
    const component = renderer.create(<PlayButton {...playButton.loading} />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders a play button when audio is loaded and it is not playing', () => {
    const component = renderer.create(<PlayButton {...playButton.paused} />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders a pause button when audio is playing', () => {
    const component = renderer.create(<PlayButton {...playButton.playing} />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
