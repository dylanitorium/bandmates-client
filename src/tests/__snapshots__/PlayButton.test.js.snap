// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayButton /> renders a pause button when audio is playing 1`] = `
<div
  className="button"
  onClick={[Function]}
>
  <i
    className="far fa-pause-circle"
  />
</div>
`;

exports[`<PlayButton /> renders a play button when audio is loaded and it is not playing 1`] = `
<div
  className="button"
  onClick={[Function]}
>
  <i
    className="far fa-play-circle"
  />
</div>
`;

exports[`<PlayButton /> renders a spinner when audio is loading 1`] = `
<div
  className="button"
  onClick={[Function]}
>
  <i
    className="fas fa-spinner fa-spin"
  />
</div>
`;
